### PR TITLE
Paginaton

### PR DESCRIPTION
**Pull request by Jahnavi-pabbathi**
Added pagination to getAllBooks endpoint

### CODE REVIEW

**Review by ISHAASTHANA — APPROVED** [review]

**Review by Pavan-KalyanT — APPROVED** [review]
working ok

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java [hunk 1]
```diff
@@ -12,18 +13,28 @@
 @RestController
 @RequestMapping("library_service_api/v1")
```
Comment (akarsh7791) [inline LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java:14]: "library/v1"

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java [hunk 2]
```diff
@@ -12,18 +13,28 @@
...
     private final LibraryService libraryService;
     public LibrarySystemController(LibraryService libraryService){
         this.libraryService=libraryService;
     }
 
-    @GetMapping("getAllBooks")
-    public List<Book> getAllBooks(){
-        return libraryService.getAllBooks();
-    }
+    @GetMapping("allBooks")
```
Comment (akarsh7791) [inline LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java:22]: "books"

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java [hunk 3]
```diff
@@ -12,18 +13,28 @@
...
+    public List<Book> getAllBooks(@RequestParam(value = "pageNumber", defaultValue = "0", required = false) Integer pageNumber) {
```
Comment (akarsh7791) [inline LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java:23]: Request parameters should also in snake case.

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java [hunk 4]
```diff
@@ -12,18 +13,28 @@
...
 
-    @PostMapping("addBook")
-    public String addBook(@RequestBody Book book){return libraryService.addBook(book);}
+            if (pageNumber < 0) {
+                return null;
+            }
+            books = libraryService.getAllBooks(pageNumber, pageSize);
+            return books;
+        } catch (Exception e) {
+            return null;
```
Comment (akarsh7791) [inline LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java:35]: Avoid returning `null`

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Repository/BookRepository.java [hunk 5]
```diff
@@ -9,6 +9,6 @@
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Integer> {
-    public List<Book> findAll();
+  //  public List<Book> findAll();
```
Comment (akarsh7791) [inline LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Repository/BookRepository.java:12]: Remove comments

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java (hunk 1 above)
Comment (ananya-anni) [reply LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java:14]: Done

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java (hunk 2 above)
Comment (ananya-anni) [reply LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java:22]: Done

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java (hunk 4 above)
Comment (ananya-anni) [reply LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java:35]: Done

File: LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java (hunk 2 above)
Comment (Jahnavi-pabbathi) [reply LibrarySystemService/src/main/java/com/target/ready/library/system/service/LibrarySystemService/Controller/LibrarySystemController.java:22]: will books_directory work??

**Pull request merged**